#!/usr/bin/python

PACKAGE="cascaded_pid_control"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
gen.add("mass", double_t, 0, "Mass of the drone.", 1.0, 0.1, 3.0)

gen.add("xy_same_params", bool_t, 0, "Use same parameters for lateral controlling. (X & Y control)", True)

# params for X
gen.add("kp_x", double_t, 0, "P gain of lateral position control in north (X control).", 1.0, 0.0, 100.0)
gen.add("kd_x", double_t, 0, "D gain of lateral position control in north (X control).", 1.0, 0.0, 100.0)

# params for Y
gen.add("kp_y", double_t, 0, "P gain of lateral position control in west (Y control).", 1.0, 0.0, 100.0)
gen.add("kd_y", double_t, 0, "D gain of lateral position control in west (Y control). ", 1.0, 0.0, 100.0)

# params for Z
gen.add("kp_z", double_t, 0, "P gain of altitude control (Z control).", 1.0, 0.0, 100.0)
gen.add("kd_z", double_t, 0, "D gain of altitude control (Z control).", 1.0, 0.0, 100.0)

gen.add("rp_same_params", bool_t, 0, "Use same parameters for roll and pitch controlling.", True)
# params for roll
gen.add("kp_roll", double_t, 0, "P gain of roll of attitude control.", 1.0, 0.0, 100)

# params for pitch
gen.add("kp_pitch", double_t, 0, "P gain of pitch of attitude control.", 1.0, 0.0, 100)

# params for yaw
gen.add("kp_yaw", double_t, 0, "P gain of yaw of yaw control.", 1.0, 0.0, 100)

exit(gen.generate(PACKAGE, "CascadedPid", "CascadedPid"))
